hub/core/tensor.py

```diff
@@ -101,11 +107,18 @@ def add_samples_to_tensor(
         else:
             # TODO: we may want to call `tobytes` on `array` and call memoryview on that. this may depend on the access patterns we
             # choose to optimize for.
-            b = memoryview(tobytes(sample))
-
+            b = tobytes(sample)
+            if not tensor_meta.get("is_compressed", False):
+                compressor = get_compressor(tensor_meta["compression"])
+                if compressor:
```

Comment:
`if compressor:` is not needed here. the get_compressor should do this check and it is already doing that.

Having this if can result in erroneous code.

Reply:
Haven't added it yet, but it should be possible to set compression to None and in that case this would be helpful.